### DISCUSSION TITLE
Inconsistent number of trials

### DISCUSSION THREAD

**Discussion by bjmeagher** [jspsych/jsPsych#1391] [Q&A]
I'm running into an odd issue with the number of trials shown to my participants. In a recent experiment, my intention was to have 100 trials per participant, but every so often, I seem to get one who is run through more. In extreme cases, the number of trials is nearly double what I intended it to be.

Have others run into similar issues? Is there a list of "usual suspects" that would cause an error like this? More than happy to chalk this up to a bug in my code, but I'm not sure where I should even be looking. My guess is that it's some kind of front-end issue, but I'd like to be able to guard against it if possible.

TIA!

**Comment by jodeleeuw** [comment]
Is it possible that participants are refreshing the experiment and starting over? Are you saving data all at once or trial-by-trial?

**Comment by bjmeagher** [comment]
I am saving data on a trial-by-trial basis. If participants are starting over, they shouldn't be recorded as the same participant once the page is refreshed; a new random subject ID is created each time.

**Comment by jodeleeuw** [comment]
Does the data you are getting give you any clues about what could be going on? For example, what does the `trial_index` column look like for the subjects with too many trials. Does it just keep going up?

How is your experiment set up to deliver 100 trials? Is it just a fixed timeline? Timeline variables? Something else?

**Reply by bjmeagher** [reply]
Sorry about the late reply on this!

I didn't keep track of the trial_index column in my database, so that info's unfortunately not available to me. I haven't noticed anything in the data that would hint at what's going on. 

My experiment delivers trials by a mixture of both timeline variables and a fixed timeline. I have a set number of stimuli for 6 blocks of 10 trials each. I include those stimuli as timeline variables for each block, but keep the blocks as separate entities in the overall timeline. I then do something similar for a final test block, which contains 40 trials.

**Comment by jodeleeuw** [comment]
Moving this over to a discussion thread
